src/wallet.rs

```diff
@@ -0,0 +1,962 @@
+    pub fn is_owned(&self) -> bool {
+        self.index.is_some()
+    }
+
+    pub(crate) fn psk(&self) -> &PublicSpendKey {
+        &self.psk
+    }
+
+    pub(crate) fn index(&self) -> &Option<u64> {
+        &self.index
+    }
+
+    /// A trimmed version of the address to display as preview
+    pub fn preview(&self) -> String {
+        let addr = bs58::encode(self.psk.to_bytes()).into_string();
+        (&addr[..10]).to_string()
+    }
+}
+
+impl FromStr for Address {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let dec = bs58::decode(s).into_vec()?;
+        if dec.len() != SEED_SIZE {
+            return Err(Error::BadAddress);
+        }
+        let mut addr_bytes = [0u8; SEED_SIZE];
+        addr_bytes.copy_from_slice(&dec);
+        let addr = Address {
+            index: None,
+            psk: dusk_pki::PublicSpendKey::from_bytes(&addr_bytes)?,
+        };
+        Ok(addr)
+    }
+}
+
+impl TryFrom<String> for Address {
+    type Error = Error;
+
+    fn try_from(s: String) -> Result<Self, Self::Error> {
+        let dec = bs58::decode(s).into_vec()?;
+        if dec.len() != SEED_SIZE {
+            return Err(Error::BadAddress);
+        }
+        let mut addr_bytes = [0u8; SEED_SIZE];
+        addr_bytes.copy_from_slice(&dec);
+        let addr = Address {
+            index: None,
+            psk: dusk_pki::PublicSpendKey::from_bytes(&addr_bytes)?,
+        };
+        Ok(addr)
+    }
+}
+
+impl TryFrom<&[u8; 64]> for Address {
+    type Error = Error;
+
+    fn try_from(bytes: &[u8; 64]) -> Result<Self, Self::Error> {
+        let addr = Address {
+            index: None,
+            psk: dusk_pki::PublicSpendKey::from_bytes(bytes)?,
+        };
+        Ok(addr)
+    }
+}
+
+impl PartialEq for Address {
+    fn eq(&self, other: &Self) -> bool {
+        self.index == other.index && self.psk == other.psk
+    }
+}
+
+impl std::hash::Hash for Address {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.index.hash(state);
+        self.psk.to_bytes().hash(state);
+    }
+}
+
+impl Display for Address {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", bs58::encode(self.psk.to_bytes()).into_string())
+    }
+}
+
+impl Debug for Address {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", bs58::encode(self.psk.to_bytes()).into_string())
+    }
+}
+
+/// Addresses holds address-related metadata that needs to be
+/// persisted in the wallet file.
+struct Addresses {
+    count: u64,
+}
+
+impl Default for Addresses {
+    fn default() -> Self {
+        Self { count: 1 }
+    }
+}
+
+impl Addresses {
+    pub(crate) fn from_bytes(buf: Vec<u8>) -> Result<Self, crate::Error> {
+        let mut u64_bytes = [0u8; 8];
+        u64_bytes.copy_from_slice(&buf);
+        Ok(Self {
+            count: u64::from_le_bytes(u64_bytes),
+        })
+    }
```

Comment:
You should implement the trait `Serializable` from `dusk_bytes` if you wan to do that, since you're using already the library in the project.

This is the only actual comment that you should fix before the merge.